learn/appendix_opengl/KHR_debug.md

```diff
@@ -0,0 +1,194 @@
+| `EnableCap.DebugOutputSynchronous` | When enabled, debug messages are sent only to the thread that generated the message. Useful when debugging and viewing the callstack. Hinders performance. |
+
+Object Label API
+----------------
+The object label API allows you to give custom labels to OpenGL objects. These
+labels are then displayed in the debug messages and graphics debuggers.
+Immensely more descriptive than the number assigned by the OpenGL context for
+each OpenGL object or a pointer.
+
+```cs
+/* Label named objects (i.e. objects with handles). */
+void GL.ObjectLabel(
+    ObjectLabelIdentifier identifier,   /* Object family, e.g. a buffer or a texture. */
+    int handle,     /* The object handle. */
+    int length,     /* Length of the label, or -1 for automatic length. */
+    string label    /* The label to give the object. */
+);
+
+/* Label pointer objects */
+void GL.ObjectPtrLabel(
+    IntPtr pointer, /* The pointer to label. */
+    int length,     /* Length of the label, or -1 for automatic length. */
+    string label    /* The label to give to the pointer. */
+);
+```
+
+Debug Group API
+---------------
+The debug group API allows you to group some commands under a label. These
+labels are then displayed by graphics debuggers in a tree fashion. This provides
+additional visual aid and context clues when reading large command lists when
+trying to figure out what went wrong.
+
+```cs
+/* Pushes a new frame onto the debug group stack. */
+void GL.PushDebugGroup(
+    DebugSourceExternal source, /* The source of the message. */
+    int id,         /* An ID to give the message. This is for your own use. */
+    int length,     /* Length of the message, use -1 for automatic length. */
+    string message  /* The label message to group following calls. */
+);
+
+/* Pops a previously pushed debug group from the stack. */
+void GL.PopDebugGroup();
+```
+
+Associated new Gets
+-------------------
+This extension has the following new gets associated with it.
+
+| Get Function | Get Enum | Initial Value | Description |
+|--------------|----------|---------------|-------------|
+| `GL.GetPointer` | `GetPointervPName.DebugCallbackFunction` | `null` | The debug callback function you provided. |
+| `GL.GetPointer` | `GetPointervPName.DebugCallbackUserParam` | `null` | The user param value you provided to the debug callback function. |
+| `GL.GetInteger` | `All.DebugLoggedMessages` | 0 | The number of messages currently in the debug log. |
+| `GL.GetInteger` | `All.DebugNextLoggedMessageLength` | 0 | Length of the next log message in queue. |
+| `GL.GetInteger` | `All.DebugGroupStackDepth` | 1 | The number of frames in the debug group stack. |
+| `GL.IsEnabled`  | `EnableCap.DebugOutput` | `false` | Is debug output enabled? |
+| `GL.IsEnabled`  | `EnableCap.DebugOutputSynchronous` | `false` | Is debug output synchronous? |
+
+Screenshots
```

Comment:
As I said before, some screenshots need to go earlier.
We can definitely have some more screenshots showing useful stuff at the end though.